Sources/StreamChat/Workers/ChannelListLinker.swift

```diff
@@ -53,7 +53,12 @@ final class ChannelListLinker {
             EventObserver(
                 notificationCenter: nc,
                 transform: { $0 as? ChannelUpdatedEvent },
-                callback: { [weak self] event in self?.unlinkChannelIfNeeded(event.channel) }
+                callback: { [weak self] event in
+                    guard let self else { return }
+                    let shouldUnlink = self.unlinkChannelIfNeeded(event.channel)
+                    guard !shouldUnlink else { return }
+                    self.linkChannelIfNeeded(event.channel)
+                }
```

Comment:
According to the documentation of `ChannelListLinker` this can break some other customers:

> /// - Channel is updated: We only check if we should remove it from the current query.
> ///   We don't try to add it to the current query to not mess with pagination.

For now, the only solution I can think of is to delegate the linking of channels to the customer when a channel is updated. 

We can either expose a closure whenever a channel is updated, or maybe better and even more flexible is to expose the linking and unlinking functions so that customers can handle linking themselves based on any event.